src/ittnotify/ittnotify_config.h

```diff
@@ -356,6 +354,17 @@ pthread_t pthread_self(void) __attribute__((weak));
 
 #endif /* ITT_PLATFORM==ITT_PLATFORM_WIN */
 
+/* strdup() is not included into C99 which results in a compiler warning about
+ * implicitly declared symbol. To avoid the issue strdup is implemented
+ * manually.
+ */
+#define __itt_fstrdup(s, new_s) do {                                        \
+    if (s == NULL) return NULL;                                             \
+    size_t s_len = __itt_fstrnlen(s, sizeof(s));                            \
```

Comment:
@igorsafo  there is a problem with "__itt_fstrnlen(s, **sizeof(s)**);" on Windows.

"sizeof(s)" is the same as "sizeof(char *)" - returns the size of the pointer, that is 8 for 64-bit machine.	

On Windows:
#define __itt_fstrnlen(s, l)      strnlen_s(s, l)
	
where "l" is max size.

Thus, **s_len** is always 8 (64-bit machine) and __itt_fstrdup will copy first 8 (64-bit machine) elements only.

Reply:
Thank you, fixed by adding a variable `max_length`.

Are there any tests you run to find these issues?

Reply:
I think it would be better to define the max length:

```
#define MAX_STRING_SIZE 4096
/* strdup() is not included into C99 which results in a compiler warning about
* implicitly declared symbol. To avoid the issue strdup is implemented
* manually.
*/
#define __itt_fstrdup(s, new_s) do {                                       \
if (s == NULL) return NULL;                                                \
size_t s_len = __itt_fstrnlen(s, MAX_STRING_SIZE);                         \
.....
```
It will help us to get rid of warning "unused variable 'max_length'" on Linux:
`#define __itt_fstrnlen(s, l)      strlen(s)`

Reply:
At this moment the repo is a mirror of the project that locates in our internal build system.
We have a plan to make a CI workflow set up to build and test the components in this repo.

Reply:
got it, updated the code. I added a prefix `ITT_STRDUP` in case the code which includes ittnotify has its own `MAX_STRING_SIZE` definition.

Reply:
good point! I'm going to merge this PR now.
I'll create a new tag for these changes tomorrow.
Thank you!